src/docs/extensions.md

```diff
@@ -2,16 +2,29 @@
 title: Extensions
 ---
 
-# Extensions
+# Extensions and Plugins
 
-Theia is composed of extensions. An extension is an npm package that exposes
-any number of DI modules (`ContainerModule`) that contribute to the creation of
-the DI container.
+Eclipse Theia is designed in a very modular and extensible way. It supports three ways of being extended/adapted to your specific requirements. These are complementary and are targeted at different use cases. You can pick the best option and even mix them within the same project.
+In the following, we will give you a quick overview of the available extension mechanisms and provide more details in the sections below.
 
-An extension is consumed by adding a dependency to the
-npm-package in the `package.json` of the application.
-Extensions can be installed/uninstalled at
-runtime, which will trigger a recompilation and restart.
+- **VS Code extensions**: Simple to write, installable at runtime, compatible with VS Code, limited to the VS Code extension API, some use cases not possible due to API restrictions => Used for adding features to existing tools. Note that you can also use existing VS Code extensions within Theia, too.
+- **Theia extensions**: Install at compile time, full access to internals of Theia via dependency injection, almost no limitations in terms of accessible API => used to build custom products and for features not covered by the VS Code extension API. Note that the Theia project (including the core) is fully built using Theia extensions in a modular way.
+- **Theia plugins**: Like VS Code extensions, additional access to some Theia specific APIs and the Frontend (frontend plugins), Theia specific parts are not compatible with VS Code.
```

Comment:
nit: `newline` after the bullet point:

```suggestion
- **Theia plugins**: Like VS Code extensions, additional access to some Theia specific APIs and the Frontend (frontend plugins), Theia specific parts are not compatible with VS Code.

```

It is confusing when looking at the rendering of the site itself, I believe this should be a separate paragraph describing the image:

![image](https://user-images.githubusercontent.com/40359487/129025559-99392a62-7495-4222-b084-ff35c8cb1a7c.png)